pkg/ui/src/Toasts/toast.ts

```diff
@@ -0,0 +1,112 @@
+import { writable, derived } from "svelte/store";
+
+import { tweened } from "svelte/motion";
+import { linear } from "svelte/easing";
+
+import { toasters } from "./toaster";
+
+// TODO: what if toast shouldn't disappear automatically?
+
+/**
+ * Sets up toast store and action
+ */
+export const consume = (_toast, target = "default") => {
+	const { toaster: toasterStore } = toasters.get(target);
+
+	// TODO: no toaster found in map...
+
+	const toastStore = createToastStore(toasterStore)(_toast);
+
+	const toastAction = createToastAction(toastStore);
+
+	return {
+		...toastStore,
+		toast: toastAction
+	};
+};
+
+/**
+ * Creates a toast store to hold and manage toast data
+ */
+export const createToastStore = (toasterStore) => (_toast) => {
+	const toastStore = writable(_toast);
```

Comment:
I don't see any types here, and am curious as to how TS didn't complain (or am I missing something?)